.vscode/settings.json

```diff
@@ -1,9 +1,25 @@
 {
-  "rust-analyzer.linkedProjects": [
-    "bindings_ffi/Cargo.toml",
-    "examples/cli/Cargo.toml",
-    "xmtp_api_grpc_gateway/Cargo.toml"
-  ],
+  "rust-analyzer": {
+    "cargo": {
+      "sysroot": "discover"
+    },
+    "linkedProjects": [
+      "bindings_ffi/Cargo.toml",
+      "examples/cli/Cargo.toml",
+      "xmtp_api_grpc_gateway/Cargo.toml"
+    ],
+    "procMacro" : {
+      "enable": true,
+      "attributes.enable": true,
+      "ignored": {
```

Comment:
Curious what issue this fixes

Reply:
Can remove, but for me with rust-analyzer, I got alot of 'unresolved-proc-macro`, for well-known macros like async-trait and others. This tells rust-analyzer to ignore them and avoid polluting the error-namespace. I think @37ng also got similar errors iirc but may be conflating it with different lsp errors.

Reply:
If it's helping your Rust analyzer and doesn't have any adverse side effects, happy to leave it as-is.